Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.h

```diff
@@ -224,6 +224,9 @@ namespace AssetProcessor
         //! Request to invalidate and reprocess a source asset or folder containing source assets
         AZ::u64 RequestReprocess(const QString& sourcePath);
         AZ::u64 RequestReprocess(const AZStd::list<AZStd::string>& reprocessList);
+        
+        AZStd::optional<AZ::s64> GetIntermediateAssetScanFolderId() const;
```

Comment:
Maybe a good comment here to explain why optional<> is preferred, like why would there not be an intermediate scan folder?

Reply:
I've added a comment. I only used optional because this is basically just a pass through to call the function on the platform config.